### PR TITLE
Add Ipmi Sensor commands Set Threshold and Get Threshold. 

### PR DESCRIPTION
**Pull request by apop5**
## Description

Added file IpmiCommandLibNetSensor.c to hold sensor related functions. 
Added functions IpmiSetSensorThreshold and IpmiGetSensorThreshold.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [x] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested
n/a

## Integration Instructions
N/A

### CODE REVIEW

**Comment by apop5** [comment]
PR is dependent on https://github.com/microsoft/mu_basecore/pull/288 completing.

File: IpmiFeaturePkg/Library/IpmiCommandLib/IpmiCommandLibNetFnSensor.c [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/** @file
+  IPMI Command - NetFnSensor.
+
+  Copyright (c) 2018 - 2021, Intel Corporation. All rights reserved.<BR>
```
Comment (cfernald) [inline IpmiFeaturePkg/Library/IpmiCommandLib/IpmiCommandLibNetFnSensor.c:4]: Is this Intel code from somewhere? if not use the MSFT copyright

Comment (apop5) [reply IpmiFeaturePkg/Library/IpmiCommandLib/IpmiCommandLibNetFnSensor.c]: Good catch, I missed that when creating the file.

**Review by cfernald — APPROVED** [review]

File: IpmiFeaturePkg/Library/IpmiCommandLib/IpmiCommandLibNetFnSensor.c [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  PDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/IpmiBaseLib.h>
+
+#include <IpmiFeature.h>
+
+/**
```
Comment (makubacki) [inline IpmiFeaturePkg/Library/IpmiCommandLib/IpmiCommandLibNetFnSensor.c:14]: I suggest updating the function descriptions to match our newer style (i.e. drop `@brief`, add `[in]`/`[out]` to params, etc.)

File: IpmiFeaturePkg/Library/IpmiCommandLib/IpmiCommandLibNetFnSensor.c [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    ResponseDataSize = sizeof (*CompletionCode);
+
+    Status = IpmiSubmitCommand (
+               IPMI_NETFN_SENSOR_EVENT,
+               IPMI_SENSOR_SET_SENSOR_THRESHOLDS,
+               (UINT8 *)SetSensorThresholdRequestData,
+               sizeof (*SetSensorThresholdRequestData),
+               (UINT8 *)&CompletionCode,
+               &ResponseDataSize
+               );
```
Comment (makubacki) [inline IpmiFeaturePkg/Library/IpmiCommandLib/IpmiCommandLibNetFnSensor.c:43]: My understanding is that `IpmiSubmitCommand()` takes a `UINT8*` for the 5th parameter (`ResponseData`):

```c
EFI_STATUS
IpmiSubmitCommand (
  IN UINT8       NetFunction,
  IN UINT8       Command,
  IN UINT8       *CommandData,
  IN UINT32      CommandDataSize,
  OUT UINT8      *ResponseData,
  IN OUT UINT32  *ResponseDataSize
  );
```

Wouldn't it just write to the `CompletionCode` buffer here?

```suggestion
    Status = IpmiSubmitCommand (
               IPMI_NETFN_SENSOR_EVENT,
               IPMI_SENSOR_SET_SENSOR_THRESHOLDS,
               (UINT8 *)SetSensorThresholdRequestData,
               sizeof (*SetSensorThresholdRequestData),
               CompletionCode,
               &ResponseDataSize
               );

```

File: IpmiFeaturePkg/Library/IpmiCommandLib/IpmiCommandLibNetFnSensor.c [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+#include <Library/IpmiBaseLib.h>
+
+#include <IpmiFeature.h>
+
+/**
+ * @brief Send the Set Sensor Threshold command for a specified SensorNumber
+ *
+ * @param SetSensorThresholdRequestData The filled out Set Sensor Threshold command structure
+ * @param CompletionCode Pointer to a buffer for returning the completion code
+ * @return EFI_STATUS
```
Comment (makubacki) [inline IpmiFeaturePkg/Library/IpmiCommandLib/IpmiCommandLibNetFnSensor.c:19]: Please add `EFI_SUCCESS` and `EFI_INVALID_PARAMETER`.

File: IpmiFeaturePkg/Library/IpmiCommandLib/IpmiCommandLibNetFnSensor.c [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+               sizeof (*SetSensorThresholdRequestData),
+               (UINT8 *)&CompletionCode,
+               &ResponseDataSize
+               );
+  }
+
+  return Status;
+}
+
+/**
```
Comment (makubacki) [inline IpmiFeaturePkg/Library/IpmiCommandLib/IpmiCommandLibNetFnSensor.c:49]: Similar comments to the other function.

**Review by makubacki — APPROVED** [review]

**Pull request merged**